crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -645,6 +647,13 @@ where
                         return Ok(invalid_fcu_response)
                     }
 
+                    // TODO(clabby): We should only do this if the chainspec has an optimism
+                    // config. Does this belong here?
+                    #[cfg(feature = "optimism")]
+                    if attrs.gas_limit.is_none() {
+                        return Ok(OnForkChoiceUpdated::invalid_payload_attributes())
+                    }
```

Comment:
@rkrasiuk Any suggestions on where to put this? In `op-geth`, the engine API itself has access to the chain config, but I can't seem to find anywhere that `BeaconConsensusEngine` has access. Ideally this check happens in the endpoint handler rather than in the payload builder

Reference in `op-geth`: [here](https://github.com/ethereum-optimism/op-geth/blob/39d121af21bc4cb45c46edf2dc1bf33daa28c54d/eth/catalyst/api.go#L342-L344)